reana_server/reana_admin/consumer.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Admin MessageConsumer."""
+
+import json
+from enum import Enum, auto
+from typing import List, Optional
+
+import click
+
+from reana_commons.consumer import BaseConsumer
+
+
+class UserDecision(Enum):
+    """Possible user decisions in interactive consumer mode."""
+
+    DELETE_MESSAGE = auto()
+    KEEP_MESSAGE = auto()
+    STOP_CONSUMER = auto()
+
+
+class MessageConsumer(BaseConsumer):
+    """Consumer responsible for cleaning queues."""
+
+    def __init__(
+        self,
+        queue_name: str,
+        key: Optional[str],
+        values_to_delete: List[str],
+        is_interactive: bool = False,
+        **kwargs,
+    ):
+        """Initialise the class."""
+        super(MessageConsumer, self).__init__(queue=queue_name, **kwargs)
+        self.key = key
+        self.values_to_delete = values_to_delete
+        self.is_interactive = is_interactive
+
+        click.secho(
+            "MessageConsumer initialized with the following settings: \n"
+            f"  - queue_name: {queue_name}\n"
+            f"  - message_key: {self.key}\n"
+            f"  - values_to_delete: {self.values_to_delete}\n"
+            f"  - interactive: {self.is_interactive}"
+        )
+
+    def get_consumers(self, Consumer, channel):
+        """Implement providing kombu.Consumers with queues/callbacks."""
+        return [
+            Consumer(
+                queues=self.queue,
+                callbacks=[self.on_message],
+                accept=[self.message_default_format],
+                prefetch_count=50,
```

Comment:
I have noticed that when the consumer is waiting for the user's decision, it might happen that multiple messages are received by the consumer. In this case, using `Ctrl+C` or replying with `stop` does not actually stop the consumer, but it will show the next message in the queue. Would `prefetch_count=1` fix this issue? Are there any cons with this approach?

Reply:
Honestly, no. It was `1` before. I will change it back.

Reply:
Works well now! 🚀